app/assets/javascripts/discourse/app/templates/components/bookmark-list.hbs

```diff
@@ -0,0 +1,79 @@
+{{#conditional-loading-spinner condition=loading}}
+  {{#load-more selector=".bookmark-list .bookmark-list-item" action=(action "loadMoreBookmarks")}}
+    <table class="topic-list bookmark-list">
+      <thead>
+        {{#if site.mobileView}}
+          <th>&nbsp;</th>
+          <th>{{i18n "topic.title"}}</th>
+          <th>&nbsp;</th>
+        {{else}}
+          <th>{{i18n "topic.title"}}</th>
+          <th>&nbsp;</th>
+          <th class="post-metadata">{{i18n "post.bookmarks.updated"}}</th>
+          <th class="post-metadata">{{i18n "activity"}}</th>
+          <th>&nbsp;</th>
+        {{/if}}
+      </thead>
+      <tbody>
+        {{#each content as |bookmark|}}
+          <tr class="topic-list-item bookmark-list-item">
+            {{#if site.mobileView}}
+              <td>
+                {{#if bookmark.post_user_avatar_template}}
+                  <a href={{bookmark.postUser.path}} data-user-card={{bookmark.post_user_username}}>
+                    {{avatar bookmark.postUser avatarTemplatePath="avatar_template" usernamePath="username" namePath="name" imageSize="small"}}
+                  </a>
+                {{/if}}
+              </td>
+            {{/if}}
+            <td class="main-link">
+              <span class="link-top-line">
+                <div class="bookmark-metadata">
+                  {{#if bookmark.name}}
+                    <span class="bookmark-metadata-item">
+                      {{d-icon "info-circle"}}{{bookmark.name}}
+                    </span>
+                  {{/if}}
+                  {{#if bookmark.reminder_at}}
+                    <span class="bookmark-metadata-item">
+                      {{d-icon "far-clock"}}{{bookmark.formattedReminder}}
+                    </span>
+                  {{/if}}
+                </div>
+
+                {{topic-status topic=bookmark}}
+                {{topic-link bookmark}}
```

Comment:
I would wrap this into something like this:

```
<div {{on "click" this.openBookmark}}>
  {{topic-link bookmark}}
</div>
```

And then you could refactor the component to only need this:

```
  @action
  openBookmark(event) {
    let link = event.target;

    if (shouldOpenInNewTab(link.href)) {
      event.preventDefault();
      openLinkInNewTab(link);
    }
  },
```

This is a little bit annoying and I would prefer if we could have a way to intercept at topic-link level, but that should do the job correctly, and is a lot less code than what you have.

Maybe a followup PR could be to improve topic-link to support opening in new tab.

Reply:
@jjaffeux the problem here is links appearing in the excerpt a few lines down, not the topic-link. The topic link is never an external link so that "open external links in new tab" preference will not affect them. However this could work for the wrapper around excerpt so I'll try that!